packages/eslint-plugin/src/rules/no-array-delete.ts

```diff
@@ -0,0 +1,84 @@
+import type { TSESLint, TSESTree } from '@typescript-eslint/utils';
+import { AST_NODE_TYPES } from '@typescript-eslint/utils';
+import type * as ts from 'typescript';
+
+import { createRule, getParserServices } from '../util';
+
+type MessageId = 'noArrayDelete' | 'useSplice';
+
+export default createRule<[], MessageId>({
+  name: 'no-array-delete',
+  meta: {
+    hasSuggestions: true,
+    type: 'problem',
+    docs: {
+      description: 'Disallow the delete operator with array expressions',
```

Comment:
[Docs] _"expression"_ isn't the right word here, it's really anything that's an array type. That can include identifiers, returned values from calls, etc.

```suggestion
      description: 'Disallow using the `delete` operator on array values',
```

For reference, https://typescript-eslint.io/rules/no-dynamic-delete currently goes with:

> `Disallow using the delete operator on computed key expressions.`

Reply:
sure, done :)